src/main/java/com/google/cloud/pubsublite/spark/PslSparkUtils.java

```diff
@@ -83,6 +95,67 @@ public static InternalRow toInternalRow(
     return InternalRow.apply(asScalaBufferConverter(list).asScala());
   }
 
+  @SuppressWarnings("unchecked")
+  private static <T> void extractVal(
+      StructType inputSchema,
+      InternalRow row,
+      String fieldName,
+      DataType expectedDataType,
+      Consumer<T> consumer) {
+    if (!inputSchema.getFieldIndex(fieldName).isEmpty()) {
+      Integer idx = (Integer) inputSchema.getFieldIndex(fieldName).get();
+      try {
+        consumer.accept((T) row.get(idx, expectedDataType));
+      } catch (ClassCastException e) {
+        // This means the field has a wrong class type.
```

Comment:
You should probably fail instead of silently swallowing this issue.

Reply:
We should not fail here, it's just a mismatch of type that we will just not use that column. Let me add a log line so customer knows why it's dropped.

Reply:
Nack. If they have a type mismatch coming into this, their schema doesn't match and they should fix it instead of silently dropping data. The risk that they will be incorrect and not notice is more important than the risk that they will have an unrelated field and be slightly annoyed they need to drop it. Please make this a hard error

Reply:
Update this so I check this in driver instead, and then in worker, it's guaranteed to match.